google-cloud-pubsub/samples/pubsub_commit_avro_schema.rb

```diff
@@ -0,0 +1,36 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "google/cloud/pubsub"
+
+def commit_avro_schema schema_id:, avsc_file:
+  # [START pubsub_commit_avro_schema]
+  # schema_id = "your-schema-id"
+  # avsc_file = "path/to/a/avsc_file.avsc"
+
+  pubsub = Google::Cloud::Pubsub.new
+  schemas = pubsub.schemas
+
+  schema = schemas.get_schema name: pubsub.schema_path(schema_id)
+
+  definition = File.read avsc_file
+  schema.definition = definition
```

Comment:
Nit: Since you don't use `definition` anywhere else, you can forgo the local variable and simply:

```ruby
schema.definition = File.read avsc_file
```

In some of the other samples below, it's maybe a bit more warranted to have the local variable since its usage is buried deeper in an expression. But in this particular case, I think it's hard to argue that the existing two lines are any more readable than the one line I suggest.

(Yes, I'm being nitpicky, but the whole purpose of a sample is to be read by humans, so cleanliness and readability are the highest priority.)